terraform/applications/discourse/email.tf

```diff
@@ -0,0 +1,276 @@
+  name    = "_amazonses.${aws_ses_domain_identity.main.id}"
+  type    = "TXT"
+  ttl     = "600"
+  records = [aws_ses_domain_identity.main.verification_token]
+}
+
+resource "aws_ses_domain_dkim" "main" {
+  domain = aws_ses_domain_identity.main.domain
+}
+
+resource "aws_route53_record" "dkim" {
+  count   = 3
+  zone_id = module.discourse_mozilla.zone_id
+  name = format(
+    "%s._domainkey.%s",
+    element(aws_ses_domain_dkim.main.dkim_tokens, count.index),
+    aws_ses_domain_identity.main.domain,
+  )
+  type    = "CNAME"
+  ttl     = "600"
+  records = ["${element(aws_ses_domain_dkim.main.dkim_tokens, count.index)}.dkim.amazonses.com"]
+}
+
+#
+# SES MAIL FROM Domain
+#
+resource "aws_ses_domain_mail_from" "main" {
+  domain           = aws_ses_domain_identity.main.domain
+  mail_from_domain = "mail.${aws_ses_domain_identity.main.domain}"
+}
+
+#
+# SES SPF validaton record & MX from Record
+#
+resource "aws_route53_record" "spf_mail_from" {
+  zone_id = module.discourse_mozilla.zone_id
+  name    = aws_ses_domain_mail_from.main.mail_from_domain
+  type    = "TXT"
+  ttl     = "600"
+  records = ["v=spf1 include:amazonses.com -all"]
+}
+
+resource "aws_route53_record" "spf_domain" {
+  zone_id = module.discourse_mozilla.zone_id
+  name    = aws_ses_domain_identity.main.domain
+  type    = "TXT"
+  ttl     = "600"
+  records = ["v=spf1 include:amazonses.com -all"]
+}
+
+resource "aws_route53_record" "mx_send_mail_from" {
+  zone_id = module.discourse_mozilla.zone_id
+  name    = aws_ses_domain_mail_from.main.mail_from_domain
+  type    = "MX"
+  ttl     = "600"
+  records = ["10 feedback-smtp.${local.workspace.region}.amazonses.com"]
+}
+
+# Receiving MX Record
+resource "aws_route53_record" "mx_receive" {
+  zone_id = module.discourse_mozilla.zone_id
+  name    = aws_ses_domain_identity.main.domain
+  type    = "MX"
+  ttl     = "600"
+  records = ["10 inbound-smtp.${local.workspace.region}.amazonaws.com"]
+}
+
+# DMARC TXT Record
+resource "aws_route53_record" "txt_dmarc" {
+  zone_id = module.discourse_mozilla.zone_id
+  name    = "_dmarc.${aws_ses_domain_identity.main.domain}"
+  type    = "TXT"
+  ttl     = "600"
+  records = ["v=DMARC1; p=reject;rua=mailto:postmaster@${aws_ses_domain_identity.main.domain};"]
+}
+
+resource "aws_s3_bucket" "incoming_email" {
+  bucket = "discourse-${local.workspace.environment}-email-${random_id.bucket.dec}"
+  acl    = "private"
+
+  lifecycle_rule {
+    enabled = true
+
+    expiration {
+      days = 21
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "incoming_email_policy" {
+  bucket = aws_s3_bucket.incoming_email.id
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AllowSESPuts",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ses.amazonaws.com"
+						},
+            "Action": "s3:PutObject",
+            "Resource": "${aws_s3_bucket.incoming_email.arn}/*"
+				}
```

Comment:
Format here looks a little weird, could be github or maybe it's doing python style formatting? Either way not a big deal

Reply:
well tf fmt passes, what are you thinking looks wrong here?

Reply:
The JSON object is formatted kind of like Python. For example in this same pr `"aws_iam_user_policy" "ses"` is formatted like JSON. Does that make sense?

Reply:
It looks like just whitespace changes, probably not too big of a deal either way, but if you were to use a jsonlinting tool it'd look something like the below, though 4 spaces like I have below seems a little suboptimal.
```
{
	"Version": "2012-10-17",
	"Statement": [{
		"Action": [
			"ses:SendRawEmail"
		],
		"Effect": "Allow",
		"Resource": "${aws_ses_domain_identity.main.arn}"
	}]
}
```

Reply:
i'm just ignoring this, sorry guys.